docs/developers-guide/driver-changelog.md

```diff
@@ -4,6 +4,18 @@ title: Driver interface changelog
 
 # Driver Interface Changelog
 
+## Metabase 0.50.0
+
+- The multimethod `metabase.driver/describe-table-fks` has been deprecated in favor of `metabase.driver/describe-fks`.
+  `metabase.driver/describe-table-fks` will be removed in 0.53.0.
```

Comment:
> 0.53.0
are you sure? I'd put in future releases if we didn't have a plan in place for 53.

Reply:
Good question. Happy discuss whether or not to remove it, but if we decide to remove it I believe we should commit to a release, rather than drawing it out. Breaking changes are bad but it's better if we do them earlier rather than later.

Here's how we could maintain backwards compatibility:
- provide a default implementation for `describe-fks` in terms of `describe-table-fks`. The implementation will select tables from the app DB in a reducible collectoin, and reduce over them, using `describe-table-fks` instead to select the FK metadata from the DB.
- use the new default implementation instead of `describe-table-fks`, unless the driver explicitly supports the `describe-fks` feature

Here's why breaking change wouldn't be too much of a hassle for driver authors:
- they could simply copy+paste the default implementation to implement `describe-fks` instead of implementing `describe-table-fks`.

In all, the pros and cons of breaking backwards compatibility aren't all that large either way. The benefit is that the interface for drivers is smaller (one less driver method and feature), so new drivers are slightly easier to implement. The downside is that all existing drivers have to copy the default implementation of `describe-fks`.

We could defer this decision until I've written the above default implementation, and we can assess the pros vs cons.

So, how do you feel about each option, between:
1. maintain backwards compatibility forever, keeping both `describe-fks` and `describe-table-fks`, but one needs to be implemented
2. deprecate in 53
3. defer this decision until the default implementation is used instead. Just say it's deprecated and may be removed

Reply:
We currently don't use milestones for this, I wonder what the current process is. I've asked here https://metaboat.slack.com/archives/CKZEMT1MJ/p1710244644500109